### PR TITLE
 download ignition data using tls config

### PR DESCRIPTION
**Pull request by SANJEEV-Choubey**

### CODE REVIEW

**Comment by relyt0925** [comment]
Solves:
TLS for ignition payload download to ignition server: https://github.com/openshift/hypershift/issues/354

File: control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go [hunk 1]
```diff
@@ -11,8 +11,10 @@ import (
 func ReconcileMachineConfigServerCert(secret, ca *corev1.Secret, ownerRef config.OwnerRef) error {
 	hostNames := []string{
 		"machine-config-server",
+		fmt.Sprintf("machine-config-server.%s", secret.Namespace),
 		fmt.Sprintf("machine-config-server.%s.svc", secret.Namespace),
 		fmt.Sprintf("machine-config-server.%s.svc.cluster.local", secret.Namespace),
+		fmt.Sprintf("*.machine-config-server.%s.svc.cluster.local", secret.Namespace),
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go:17]: do we really need the others domains?

Comment (SANJEEV-Choubey) [reply control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go:17]: Handling request using "*.machine-config-server.%s.svc.cluster.local", other one kept it just for failsafe But can be removed.

**Comment by enxebre** [comment]
lgtm, just need to squash commits and run tests before tagging
/hold

File: control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go (hunk 1 above)
Comment (relyt0925) [reply control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go:17]: we can remove the others that's fine just need the final wildcard one

Comment (SANJEEV-Choubey) [reply control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go:16]: This has been fixed

Comment (enxebre) [reply control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go:16]: >This has been fixed

I can still see all of them in this diff

**Comment by enxebre** [comment]
@SANJEEV-Choubey the commit description contains confusing [references to your local development](https://github.com/openshift/hypershift/pull/453/commits/f64e92fba189df437a168da31c0cc66ab16a37ee). Please remove those.
FWIW To arrange development commits once you are ready you can leverage `git rebase -i`, `git ci --amend` and `git push -f`. Please note that `git push -f` should be used with extreme caution.

Please update the GH description and the git commit description with the same content, i.e `git ci -m"Imperative mood change" -m"Why/how/context"` e.g "To enable tls for downloading ignition this creates a headless service. The additional domains for the certs are also removed as they are not needed."

**Comment by SANJEEV-Choubey** [comment]
@enxebre could you please review https://github.com/openshift/hypershift/pull/457 , I have created same PR  with single commit and the proper commit message. I am closing this PR.

**Comment by enxebre** [comment]
hey @SANJEEV-Choubey thanks, there was no need to close this one but just to rewrite your local commit history as in https://github.com/openshift/hypershift/pull/453#issuecomment-910080382. I'll have a look at 457.

**Pull request closed without merging**